VideoLocker/src/main/java/org/edx/mobile/model/api/CourseEntry.java

```diff
@@ -248,22 +252,80 @@ public boolean isGroupAvailable(SocialUtils.SocialType type) {
 
     }
 
-    public String getDescription(Context context){
+    public String getDescription(Context context, boolean withStartDate) {
         StringBuilder detailBuilder = new StringBuilder();
-        if ( getOrg() != null){
-            detailBuilder.append( getOrg());
+
+        if (!TextUtils.isEmpty(org)) {
+            detailBuilder.append(org);
         }
-        if ( getNumber() != null) {
-            if (detailBuilder.length() > 0){
+
+        if (!TextUtils.isEmpty(number)) {
+            if (detailBuilder.length() > 0) {
                 detailBuilder.append(" | ");
             }
-            detailBuilder.append( getNumber());
+            detailBuilder.append(number);
 
         }
-        //https://openedx.atlassian.net/browse/MA-858
-        //we remove the ending data for now
+
+        if (withStartDate) {
+            if (detailBuilder.length() > 0) {
+                detailBuilder.append(" | ");
+            }
+
+            detailBuilder.append(getFormattedStartDate(context));
+        }
+
+        if (detailBuilder.length() > 0) {
+            int secondlastCharIndex = detailBuilder.length() - 2;
+            if (detailBuilder.charAt(secondlastCharIndex) == '|') {
+                detailBuilder.delete(secondlastCharIndex, detailBuilder.length());
+            }
+        }
 
         return detailBuilder.toString();
     }
 
+    @SuppressLint("SimpleDateFormat")
+    public String getFormattedStartDate(Context context){
+        StringBuilder dateBuilder = new StringBuilder();
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("MMMM dd");
```

Comment:
@aleffert: The default constructor already delegates to the other one passing `Locale.getDefault()`, which is actually the very reason why this warning is shown in that case, as `SimpleDateFormat` doesn't support date localization as one would normally assume, but pattern localization (in which case you would want to hard-code the pattern). As explained in the [Android docs](http://developer.android.com/reference/java/text/SimpleDateFormat.html):

> The main reason you'd create an instance this class directly is because you need to format/parse a specific machine-readable format, in which case you almost certainly want to explicitly ask for [`US`](http://developer.android.com/reference/java/util/Locale.html#US) to ensure that you get ASCII digits (rather than, say, Arabic digits). (See "[Be wary of the default locale](http://developer.android.com/reference/java/util/Locale.html#default_locale)".) The most useful non-localized pattern is `"yyyy-MM-dd HH:mm:ss.SSSZ"`, which corresponds to the ISO 8601 international standard date format.

Therefore, in API 18+ we can use the above-mentioned `getBestDateTimePattern()` method to get a localized form of a specific format. However in lower API levels, we need to make a choice between either using the existing `getDateInstance()`, which would return a localized date without any specified pattern, or using a completely non-localized date pattern via `SimpleDateFormat`.